cmd/privateendpoints-setuplink.go

```diff
@@ -142,6 +148,43 @@ var privateEndpointsSetupLinkCmd = &cobra.Command{
 			if err != nil {
 				logger.Fatal("failed to enable private dns", zap.Error(err))
 			}
+		} else if instanceIdGcp != "" {
+			if !config.GCP.Enabled.Value() {
+				logger.Fatal("cannot setup GCP private endpoint without GCP configuration")
+			}
+
+			if projectIdGcp == "" {
+				logger.Fatal("must specify gcp-project-id when using gcp-instance-id")
+			}
+
+			gcpCreds := helper.GetGCPCredentials(ctx)
+
+			peCtrl := gcpcontrol.PrivateEndpointsController{
+				Logger:    logger,
+				Zone:      config.GCP.Zone,
+				Creds:     gcpCreds,
+				ProjectID: projectIdGcp,
+			}
+
+			vpcInfo, err := peCtrl.GetNetworkAndSubnet(ctx, instanceIdGcp)
+			if err != nil {
+				logger.Fatal("failed to get network and subnet for GCP instance", zap.Error(err))
+			}
+
+			command, err := cloudDeployer.GenPrivateEndpointLinkCommand(ctx, cloudCluster.ClusterID, &capellacontrol.PrivateEndpointLinkRequest{
+				VpcID:     vpcInfo.NetworkID,
+				SubnetIds: vpcInfo.SubnetworkID,
+			})
+
+			err = helper.ExecuteBashCommand(command)
```

Comment:
We intentionally avoid directly invoking other system commands from dinocluster, instead opting to implement them directly in cbdinocluster in order to ensure that cbdinocluster remains compatible with the widest variety of platforms, to be able to provide more nuanced diagnostic information when things go wrong as well as ensuring that dinocluster does not have unnecessary dependancies for users on third-party tools.  The information needed to bind the private endpoint should be available from the API separate from the generated command they give you, if it does not currently do this, we should file an AV bug, since this will likely impact other automated tools like terraform and parse the command they provide in the short-term.